tests/unit/test_pipeline_graph.py

```diff
@@ -575,40 +578,53 @@ def test_merge_combined_implementations_parallel(default_config_params, test_dir
     check_nodes_and_edges(pipeline_graph, expected_nodes, expected_edges)
 
 
-def test_cycle_error(default_config_params) -> None:
```

Comment:
These three tests are identical and the two new ones are as well aside from needing to pass in a custom PipelineSchema - hence the refactor to a single parametrized test.